### ISSUE TITLE
removeQuery with value filter doesn't work when the parameter value is not plural / an array / collection.

### ISSUE THREAD

**Issue by ryanelian** [medialize/URI.js#250]
Example:

```
var uri = 'http://localhost:2039/product?t=1&q=test';
var uriObj = new URI(uri);
uriObj.removeQuery({t : 1});
return uriObj.href();
```

Expected returned value:

`http://localhost:2039/product?q=test`

Actual returned value:

`http://localhost:2039/product?t=1&q=test`

How I tackled the problem atm:

```
removeTagQuery: function (uri, tagId) {
    var uriObj = new URI(uri);
    var uriQueries = URI.parseQuery(uriObj.query());

    //hack for https://github.com/medialize/URI.js/issues/250
    //removeQuery with value filter doesn't work when the parameter value is not plural / an array / collection. #250 
    if (Array.isArray(uriQueries.t))
    {
        uriObj.removeQuery({
            t: tagId
        });
    } else
    {
        uriObj.removeQuery('t');
    }
        
    return uriObj.href();
}
```

**Comment by rodneyrehm** [comment]
I'm not sure I follow the plural / array / collection thought.

The problem is the *type of the value* you're trying to remove. To URI.js all query string values are *strings*.

```js
var uri = "http://localhost:2039/product?t=1&q=test";

URI(uri).removeQuery({t : 1}).toString();
// "http://localhost:2039/product?t=1&q=test"

URI(uri).removeQuery({t : "1"}).toString();
// "http://localhost:2039/product?q=test"
```

I admit URI.js could've done a better job with casting the value to string itself. To fix that, you'd sinply check for `data[name] === String(value)` [here](https://github.com/medialize/URI.js/blob/gh-pages/src/URI.js#L780). Extending the [tests](https://github.com/medialize/URI.js/blob/gh-pages/test/test.js#L851-L852) is a good idea and please open the PRs against *master*.

**Comment by ryanelian** [comment]
Very interesting insight.

I originally thought it was array removal problem because `URI(uri).removeQuery({t : 1}).toString();` works on `http://localhost:2039/product?t=1&t=2&t=3` for instance (Results in `http://localhost:2039/product?t=2&t=3`), but not when the parameter is not plural.

**Comment by rodneyrehm** [comment]
To resolve arrays [`filterArrayValues()`](https://github.com/medialize/URI.js/blob/gh-pages/src/URI.js#L98) is used, which implictly [casts all values to string](https://github.com/medialize/URI.js/blob/gh-pages/src/URI.js#L104-L110) (because property keys are strings…) - that's why it worked for `?t=1&t=2`.

Will you prepare a PR?

**Comment by ryanelian** [comment]
Sorry for the wait! I just got home. Made the PR.

Thanks in advance!

**Comment by rodneyrehm** [comment]
fixed by https://github.com/medialize/URI.js/pull/252

**Comment by rodneyrehm** [comment]
released in [v1.17.0](https://github.com/medialize/URI.js/releases/tag/v1.17.0)
